Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/ChallengesRepository.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Domain.Entities;
+using Backbone.Modules.Devices.Infrastructure.Persistence.Database;
+using Microsoft.EntityFrameworkCore;
+
+namespace Backbone.Modules.Devices.Infrastructure.Persistence.Repository;
+public class ChallengesRepository : IChallengesRepository
+{
+    private readonly DbSet<Challenge> _challenges;
+    private readonly IQueryable<Challenge> _readonlyChallenges;
+    private readonly DevicesDbContext _dbContext;
+
+    public ChallengesRepository(DevicesDbContext dbContext)
+    {
+        _challenges = dbContext.Challenges;
+        _readonlyChallenges = dbContext.Challenges.AsNoTracking();
+        _dbContext = dbContext;
+    }
+
+    public async Task<Challenge> FindById(string id, CancellationToken cancellationToken, bool track = false)
+    {
+        return await (track ? _challenges : _readonlyChallenges)
+            .Where(c => c.Id == id).FirstAsync(cancellationToken);
```

Comment:
`FirstAsync` throws an `InvalidOperationException` when no item was found, which leads to an http status code 500.

Reply:
Should I use `FirstOrDefaultAsync`?

Reply:
Yep